protocol/lavasession/provider_session_manager.go

```diff
@@ -263,6 +264,44 @@ func (psm *ProviderSessionManager) UpdateEpoch(epoch uint64) {
 	psm.currentEpoch = epoch
 	psm.consumerPairedWithProjectMap = filterOldEpochEntries(psm.blockedEpochHeight, psm.consumerPairedWithProjectMap)
 	psm.sessionsWithAllConsumers = filterOldEpochEntries(psm.blockedEpochHeight, psm.sessionsWithAllConsumers)
+	psm.latestVirtualEpoch = 0
+}
+
+func (psm *ProviderSessionManager) UpdateVirtualEpoch(epoch uint64, virtualEpoch uint64) {
+	if virtualEpoch == 0 {
```

Comment:
if we get same
atmomic.Load(psm.latestVirtualEpoch) == virtualEpoch  
lets print an error?